omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -109,12 +126,58 @@ public Object create(SimpleObject post, RequestContext context) throws ResponseE
 				case UNLOAD:
 					unloadModules(modules, servletContext);
 					break;
+				case INSTALL:
+					Module module = installModule(modules, installUri, servletContext);
+					List<Module> listModule = new ArrayList<Module>();
+					listModule.add(0, module);
+					action.setModules(listModule);
+					break;
 			}
 		}
 		
 		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
 	}
 	
+	private Module installModule(Collection<Module> modules, String installUri, ServletContext servletContext) {
+		List<Module> moduleList = new ArrayList<Module>(modules);
+		Module existingModule = moduleList.get(0);
+		Module tempModule = null;
+		File moduleFile = null;
+		
+		try {
+			//Module existingModule = moduleFactoryWrapper.getModuleById(module.getModuleId());
+			if (existingModule != null) {
+				List<Module> dependentModulesStopped = moduleFactoryWrapper.stopModuleAndGetDependent(existingModule);
+				
+				for (Module depMod : dependentModulesStopped) {
+					moduleFactoryWrapper.stopModuleSkipRefresh(depMod, servletContext);
+				}
+				
+				moduleFactoryWrapper.stopModuleSkipRefresh(existingModule, servletContext);
+				moduleFactoryWrapper.unloadModule(existingModule);
+			}
+			
+			URL downloadUrl = new URL(installUri);
+			String fileName = FilenameUtils.getName(downloadUrl.getPath());
+			InputStream inputStream = ModuleUtil.getURLStream(downloadUrl);
+			moduleFile = ModuleUtil.insertModuleFile(inputStream, fileName);
+			tempModule = moduleFactoryWrapper.loadModule(moduleFile);
+			moduleFactoryWrapper.startModule(tempModule, servletContext);
+			return tempModule;
+		}
+		catch (MalformedURLException e) {
+			throw new RuntimeException(e.getMessage());
+		}
+		catch (IOException e) {
+			throw new RuntimeException(e.getMessage());
+		}
+		finally {
+			if (tempModule == null && moduleFile != null) {
```

Comment:
Since it is moduleFile that you are deleting, isn't it enough to just check for it alone?